funds-manager/funds-manager-server/src/custody_client/gas_wallets.rs

```diff
@@ -0,0 +1,45 @@
+//! Handlers for gas wallet operations
+
+use ethers::{
+    signers::{LocalWallet, Signer},
+    utils::hex::ToHexExt,
+};
+use rand::thread_rng;
+use tracing::info;
+
+use crate::{error::FundsManagerError, helpers::create_secrets_manager_entry_with_description};
+
+use super::CustodyClient;
+
+impl CustodyClient {
+    /// Create a new gas wallet
+    pub(crate) async fn create_gas_wallet(&self) -> Result<String, FundsManagerError> {
+        // Sample a new ethereum keypair
+        let keypair = LocalWallet::new(&mut thread_rng());
+        let address = keypair.address().encode_hex();
+
+        // Add the gas wallet to the database
+        self.add_gas_wallet(&address).await?;
+
+        // Store the private key in secrets manager
+        let secret_name = Self::gas_wallet_secret_name(&address);
+        let private_key = keypair.signer().to_bytes();
+        let secret_value = hex::encode(private_key);
+        let description = "Gas wallet private key for use by Renegade relayers";
```

Comment:
nit: maybe we should include the wallet address in the description?

Reply:
it's in the name